logfire/_internal/main.py

```diff
@@ -1999,14 +2000,17 @@ def __enter__(self) -> LogfireSpan:
     def __exit__(self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: Any) -> None:
         if self._token is None:  # pragma: no cover
             return
+        assert self._span is not None
 
         context_api.detach(self._token)
         self._token = None
-
-        assert self._span is not None
-        _exit_span(self._span, exc_value)
-
-        self.end()
+        if self._span.is_recording():
+            with handle_internal_errors():
+                if self._added_attributes:
+                    self._span.set_attribute(
+                        ATTRIBUTES_JSON_SCHEMA_KEY, attributes_json_schema(self._json_schema_properties)
+                    )
```

Comment:
I just moved the logic from `end()` here, noting we can drop a check since we already know `self._span is not None`.